in_toto/attestation/v1/resource_descriptor.go

```diff
@@ -0,0 +1,53 @@
+package v1
+
+import (
+	"fmt"
+
+	ita1 "github.com/in-toto/attestation/go/v1"
+	"google.golang.org/protobuf/types/known/structpb"
+)
+
+/*
+GenerateValidResourceDescriptor constructs an in-toto Attestation Framework v1
+ResourceDescriptor struct. If the created object does not represent a
+valid ResourceDescriptor, this function returns an error.
+*/
+func GenerateValidResourceDescriptor(name string, uri string, digestSet map[string]string, content []byte, downloadLocation string, mediaType string, annotations *structpb.Struct) (*ita1.ResourceDescriptor, error) {
+	rd := &ita1.ResourceDescriptor{
+		Name:             name,
+		Uri:              uri,
+		Digest:           digestSet,
+		Content:          content,
+		DownloadLocation: downloadLocation,
+		MediaType:        mediaType,
+		Annotations:      annotations,
+	}
+
+	err := rd.Validate()
+	if err != nil {
+		return nil, fmt.Errorf("Invalid resource descriptor: %w", err)
+	}
+
+	return rd, nil
+}
+
+/*
+RDListFromRecord converts a map of artifacts as collected
+by RecordArtifacts() in runlib.go and converts it into a list
```

Comment:
Can this be expanded with the structure of the map? We may retire RecordArtifacts at some point, it'll be helpful to track this here.

Reply:
Just to be 100% sure I've understood what you're asking, you're suggesting we include `map[string]map[string][string]` in this comment?